haystack/document_store/milvus2x.py

```diff
@@ -0,0 +1,691 @@
+
+                vector_id_map = {}
+                for vector_id, doc in zip(mutation_result.primary_keys, document_batch):
+                    vector_id_map[doc.id] = str(vector_id)
+
+                self.update_vector_ids(vector_id_map, index=index)
+                progress_bar.set_description_str("Documents Processed")
+                progress_bar.update(batch_size)
+
+        # TODO: Equivalent in 2.0?
+        # self.milvus_server.compact(collection_name=index)
+
+    def query_by_embedding(self,
+                           query_emb: np.ndarray,
+                           filters: Optional[dict] = None,
+                           top_k: int = 10,
+                           index: Optional[str] = None,
+                           return_embedding: Optional[bool] = None) -> List[Document]:
+        """
+        Find the document that is most similar to the provided `query_emb` by using a vector similarity metric.
+
+        :param query_emb: Embedding of the query (e.g. gathered from DPR)
+        :param filters: Optional filters to narrow down the search space.
+                        Example: {"name": ["some", "more"], "category": ["only_one"]}
+        :param top_k: How many documents to return
+        :param index: (SQL) index name for storing the docs and metadata
+        :param return_embedding: To return document embedding
+        :return:
+        """
+        index = index or self.index
+        try:
+            from pymilvus import connections
+            from pymilvus.client.abstract import QueryResult
+        except:
+            raise ImportError("Install pymilvus===2.0.0rc6 version")
+
+        connection = connections.get_connection()
+        has_collection = connection.has_collection(collection_name=index)
+        if not has_collection:
+            raise Exception("No index exists. Use 'update_embeddings()` to create an index.")
+
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        connection.load_collection(index)
+
+        query_emb = query_emb.reshape(1, -1).astype(np.float32)
+
+        dsl: Dict[str, Any] = {
+            "bool": {
+                "must": [
+                    {
+                        "vector": {
+                            self.embedding_field: {
+                                "metric_type": self.metric_type,
+                                "params": self.search_param,
+                                "query": query_emb.tolist(),
+                                "topk": top_k
+                            }
+                        }
+                    }
+                ]
+            }
+        }
+
+        if filters is not None:
+            for k, v in filters.items():
+                dsl["bool"]["must"].append(
+                    {
+                        "term": {
+                            k: v
+                        }
+                    }
+                )
+
```

Comment:
Interesting! their dsl seems to be very elasticsearch-like :) 

Did you test these filtered queries already? 

If I followed their status correctly, the filtered queries will currently only work on a few datatypes (ints? dates?)  but not on strings. Shall we add a check here and raise an exception for string filters?